### PR TITLE
feat(sdk-coin-canton): add CantonCommand transaction support

### PR DESCRIPTION
**Pull request by abhijeet848**
## Summary

Adds SDK support for the `cantonCommand` intent, which allows callers to submit arbitrary DAML CreateCommand or ExerciseCommand transactions

### sdk-core
- New types: `CantonCommand`, `CantonCreateCommand`, `CantonExerciseCommand`, `CantonCommandParams`, `CantonCommandResolveContractSpec`
- `TransactionType.CantonCommand` enum value
- `cantonCommandParams` added to `PrebuildTransactionWithIntentOptions` and `PopulatedIntent`
- `mpcUtils`: validates `cantonCommand` intent — enforces canton coin family, non-null plain-object command with exactly one of CreateCommand/ExerciseCommand, and non-empty actAs; excludes from recipients requirement

### sdk-coin canton
- `CantonCommandBuilder`: builds the prepare-command request sent to IMS
- `Transaction`: multi-party signature support — emits one `partySignatures` entry per actAs party; restores actAs from the prepared transaction on deserialise
- `verifyTransaction`: deep verification of on chain prepared transaction against user-supplied params,  templateId (module+entity), choice, actAs, acting-party privilege check, contractId, and recursive argument comparison with resolveContracts skip-path support
- New utils: `extractCantonCommandInfo`, `extractSubmitterActAs`, `damlValueToJson`, `assertDeepCantonMatch`, `normalizeInjectAs`, `parseCantonTemplateId`, `sameElements`

### Testing
- `verifyTransaction` tests covering all success and error paths for CantonCommand (templateId mismatch, choice mismatch, contractId mismatch, actAs mismatch, argument mismatch, resolveContracts skip-path, both-commands rejection)
- Full unit coverage for all new utils functions

Ticket: SCAAS-9316

### CODE REVIEW

**Comment by zeeshanamjad-eng** [comment]
@claude

**Comment by ravibitgo** [comment]
@claude  Please do a thorough review of this PR.

Focus especially on:

- any bugs or incorrect logic in the changes
- whether these changes could impact existing flows or cause regressions
- whether the implementation is aligned with the requirement/design
- whether the changes are correct and complete
- missing edge cases, validation gaps, and test coverage gaps
- Please call out concrete issues clearly and separate them by severity if possible.

File: modules/sdk-core/src/bitgo/utils/mpcUtils.ts [hunk 1]
```diff
@@ -146,6 +146,31 @@ export abstract class MpcUtils {
...
+          cmd !== undefined &&
+          typeof cmd === 'object' &&
+          !Array.isArray(cmd) &&
+          ('CreateCommand' in cmd || 'ExerciseCommand' in cmd),
+        `'cantonCommandParams.command' must be an object with a CreateCommand or ExerciseCommand key`
+      );
+      assert(
+        !('CreateCommand' in cmd && !!cmd['CreateCommand'] && 'ExerciseCommand' in cmd && !!cmd['ExerciseCommand']),
+        `'cantonCommandParams.command' must contain exactly one of CreateCommand or ExerciseCommand, not both`
+      );
```
[the remark this answers is not in the document]
Comment (abhijeet848) [reply modules/sdk-core/src/bitgo/utils/mpcUtils.ts:166]: Fixed

**Review by ravibitgo — APPROVED** [review]

**Review by zeeshanamjad-eng — APPROVED** [review]
